store/iavl/store.go

```diff
@@ -147,6 +147,15 @@ func (st *Store) LastCommitID() types.CommitID {
 	}
 }
 
+// PausePruning implements CommitKVStore interface.
+func (st *Store) PausePruning(pause bool) {
+	if pause {
+		st.tree.SetCommitting()
+	} else {
+		st.tree.UnsetCommitting()
```

Comment:
I looked into the iavl impl and this method is not safe to be called (multiple times) without a `SetCommitting` before. This is risky for a public method as people may use it without deeper knowledge. `PausePruning(false)` does not look like a 💣 
 IMHO it makes sense to fix UnsetCommitting.

Reply:
I was also thinking if we need to provide better locking so that no other process accidentally unlocks? Do you think such a scenario exists? In this case, control must be within the iavl and you can have a `tree.DoWithCommitPaused{func(){...})` that is guarded